File: 0205. Isomorphic Strings/Solution0205.py
class Solution0205:
    def isIsomorphic(self, s: str, t: str) -> bool:
        stt = [-1] * 256        
        tts = [-1] * 256
        n = len(s)
        for i in range(n):
            if stt[ord(s[i])] == -1 and tts[ord(t[i])] == -1:
                stt[ord(s[i])] = ord(t[i])
                tts[ord(t[i])] = ord(s[i])
            elif stt[ord(s[i])] != ord(t[i]) or tts[ord(t[i])] != ord(s[i]):
                return False
        return True

File: 0205. Isomorphic Strings/Solution0205.cpp
#include<string>
#include<vector>

using namespace std;

class Solution0205 {
public:
    bool isIsomorphic(string s, string t) {
        vector<int> stt(256,-1);        
        vector<int> tts(256,-1); 
        int n = s.length();
        for (int i = 0; i < n; i++) {
            char cs = s[i], ct = t[i];
            if (stt[cs] == -1 && tts[ct] == -1) {
                stt[cs] = ct;
                tts[ct] = cs;
            } else if (stt[cs] != ct || tts[ct] != cs) {
                return false;
            }
        }
        return true;
    }
};